packages/flutter/lib/src/widgets/scroll_controller.dart

```diff
@@ -109,6 +111,7 @@ class ScrollController {
   /// controller will not manipulate the given position.
   void detach(ScrollPosition position) {
     assert(_positions.contains(position));
+    position.removeListener(notifyListeners);
```

Comment:
also do this in dispose().

...and oy, vey, we may have to ensure anyone who creates a ScrollController disposes it...